language-server/modules/langserver-compiler/src/main/java/org/ballerinalang/langserver/compiler/workspace/WorkspaceDocumentManagerImpl.java

```diff
@@ -90,14 +97,39 @@ public void openFile(Path filePath, String content) throws WorkspaceDocumentExce
      * {@inheritDoc}
      */
     @Override
-    public void updateFile(Path filePath, String updatedContent) throws WorkspaceDocumentException {
+    public void updateFile(Path filePath, List<TextDocumentContentChangeEvent> changeEvent)
+            throws WorkspaceDocumentException {
         if (isFileOpen(filePath)) {
```

Comment:
Shall we try early return here? May be for the inner if-else as well we can try that?

Reply:
+1 will update